### ISSUE TITLE
speaker diarization on mobile devices.

### ISSUE THREAD

**Issue by alamnasim** [pyannote/pyannote-audio#462]
Hi @hbredin, 

I want to run offline speaker diarization on mobile devices  (android and ios).

I am already running speaker diarization on server side and its giving me good accuracy, so Thanks a lot for that.

Now I want to run offline speaker diarization on mobile devices(android and ios).

The steps I want to follow:
1. convert sad,scd,emb models from pytorch to ONNX and then to tensorflow(tflite).
2. load audio data using essentia or sox in c++.
3. load converted model on mobile device using tflite and do inference.

My concern is:
1. Is it feasible to load pyannote-audio models in tflite if I dont have installed dependencies?
2. Can I make pyannote-audio mdodel as simple end to end model to run in any other environment (except python)?
3. Has anyone tried this on mobile device? If yes then please suggest the steps.

Any help appreciated.

Thanks

**Comment by mogwai** [comment]
I haven't done this before but I'm quite interested to see how you get on. I'd thought it would be easier to follow the pytorch mobile route: https://pytorch.org/mobile/home/.

**Comment by hbredin** [comment]
I have no experience either and this is not (for now) the priority of this project.

But, I'd happily discuss collaboration opportunities with your company (or is it a personal project of yours?)
Google me and you should be able to find my email address...

**Comment by alamnasim** [comment]
Thanks for your interest.

Sure I will ping you when I will start working on this using pyannote-audio.

Currently I don't have much time to do it from scratch or modify to make use of pyannote-audio for mobile devices.

I am looking for a pre-trained build or any opensource implementation for mobile device. I will be happy if I can work with you to make it for mobile device in future. It would be my pleasure.



Thanks

**Comment by csukuangfj** [comment]
@alamnasim 

We have ported the pyannote-audio [speaker segmentation model](https://huggingface.co/pyannote/segmentation-3.0) to Android for speaker diarization.

You can find pre-built Android APKs at
https://k2-fsa.github.io/sherpa/onnx/speaker-diarization/android.html

Everything is [open-sourced](https://github.com/k2-fsa/sherpa-onnx). It provides not only APIs for Kotlin and Java, but also APIs for more than 9 other languages, e.g., C++, Python, C, Go, Swift, Dart, C#, JavaScript, etc.

**Comment by BrandonWeng** [comment]
For anyone else looking for an iOS/macOS speaker diarization solution, we converted the PyTorch models to CoreML so that they can run on ANE.

Here's the Swift package: https://github.com/FluidInference/FluidAudio

Fully open source.
